perun-core/src/main/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntry.java

```diff
@@ -1128,7 +1162,10 @@ public Facility getFacilityForHost(PerunSession sess, Host host) throws Internal
 		if (!facilities.isEmpty()) {
 			Iterator<Facility> facilityByHostname = facilities.iterator();
 			while(facilityByHostname.hasNext()) {
-				if(!AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, facilityByHostname.next())) facilityByHostname.remove();
+				if(!AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, facilityByHostname.next()) &&
```

Comment:
This whole check is fundamentally wrong. We should return all facilities for PERUNADMIN/PERUNOBSERVER and filter response only for FACILITYADMIN.